pm_icecon/bt/compute_bt_ic.py

```diff
@@ -189,50 +198,54 @@ def _within_plusminus_10(target_value, value) -> bool:
     return wtp_set_tuple
 
 
-def ret_linfit(
+def get_linfit(
     *,
     land_mask: npt.NDArray[np.bool_],
     tb_mask: npt.NDArray[np.bool_],
-    tbx,
-    tby,
+    tbx: npt.NDArray,
+    tby: npt.NDArray,
     lnline: Line,
-    add,
-    lnchk=1.5,
-    weather_mask,
+    add: float,
+    weather_mask: npt.NDArray[np.bool_],
+    # If the calculated slope is larger than `max_slope`, a `BootstrapAlgError`
+    # is raised.
+    max_slope: float = 1.5,
     # If any one of the rest of these arguments is given, the rest must also be
     # non-None. Currently only used for getting the v1937 line, in determining
     # if pixels are valid for use.
-    tba=None,
+    tba: npt.NDArray | None = None,
     iceline: Line | None = None,
-    adoff=None,
+    ad_line_offset: float | None = None,
 ) -> Line:
-    # Reproduces both ret_linfit1() and ret_linfit2()
+    """Reproduce both `ret_linfit1()` and `ret_linfit2()` from GSFC code."""
     not_land_or_masked = ~land_mask & ~tb_mask
-    if tba is not None and iceline is not None and adoff is not None:
-        is_tba_le_modad = tba <= (tbx * iceline['slope']) + iceline['offset'] - adoff
+    # tba is always tb_h37, which is the x-axis of the 37h37v tbset.
+    # The iceline is always the v19v37 lnline (ln == linear?).
+    # ad_line_offset is calculated from the 37v37h tbset.
+    if tba is not None and iceline is not None and ad_line_offset is not None:
+        is_tba_le_modad = (
+            tba <= (tbx * iceline['slope']) + iceline['offset'] - ad_line_offset
+        )
     else:
         is_tba_le_modad = np.full_like(not_land_or_masked, fill_value=True)
 
     is_tby_gt_lnline = tby > (tbx * lnline['slope']) + lnline['offset']
 
     is_valid = not_land_or_masked & is_tba_le_modad & is_tby_gt_lnline & ~weather_mask
 
-    icnt = np.sum(np.where(is_valid, 1, 0))
-    if icnt <= 125:
-        raise BootstrapAlgError(f'Insufficient valid linfit points: {icnt}')
-
-    xvals = tbx[is_valid].flatten()
-    yvals = tby[is_valid].flatten()
+    num_valid_pixels = is_valid.sum()
```

Comment:
Is this the canonical way to count the number of "True" values in a Boolean array?

Reply:
This is typically what I do, I find it to be a bit more clear that using `np.where`. The `sum` operator casts to numeric, and for boolean `True` is 1 and `False` is 0. It's the same as writing `is_valid.astype(int).sum()`.

Reply:
I actually like the explicit casting to int better than both the sum() of the where().  You never know when someone's going to change things to be more pedantic and disallow the "sum" of boolean values.